src/backend/distributed/utils/maintenanced.c

```diff
@@ -257,6 +263,13 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		CHECK_FOR_INTERRUPTS();
 
 		/*
+		 * We clear the metadata cache before every iteration because otherwise
+		 * it might contain stale OIDs if deadlock detection runs during a DROP
+		 * EXTENSION followed by a CREATE EXTENSION.
+		 */
+		ClearMetadataCache();
```

Comment:
> Why do you think this is required 

@marcocitus's above [comment](https://github.com/citusdata/citus/pull/1584#issuecomment-322817270) explains it.

We realized this while running consecutive `DROP extension & CREATE extension` commands as [here](https://gist.github.com/onderkalaci/b69d3ec169d2b52a207bd2ded011896a), we'd see some   `ERROR:  could not open relation with OID 449464` kind of errors 

> and correct to do this here 

Well, it's not wrong but has some performance drawbacks :)